src/dataframerow/dataframerow.jl

```diff
@@ -189,6 +189,11 @@ Base.convert(::Type{Vector{T}}, dfr::DataFrameRow) where T =
 Base.Vector(dfr::DataFrameRow) = convert(Vector, dfr)
 Base.Vector{T}(dfr::DataFrameRow) where T = convert(Vector{T}, dfr)
 
+Base.convert(::Type{Array}, dfr::DataFrameRow) = Vector(dfr)
```

Comment:
This is a bit more controversial as `DataFrameRow` could also be viewed as a single-row matrix. Maybe let's leave it out for now?

BTW, we define `Base.size(r::DataFrameRow) = (length(index(r)),)`, but isn't this at odds with how broadcasting works for `DataFrameRow`? I.e. it could also be `(1,length(index(r)))`.

Reply:
We allow `Vector` to be called on  `DataFrameRow`. Since we do this then all other interpretations are ruled out I think. Doing `df[row, cols]` drops one level of dimensionality from a data frame to make it 1-dimensional object just like in arrays.

Note that we do not allow:
```
julia> using DataFrames

julia> df = DataFrame(ones(2,3))
2×3 DataFrame
│ Row │ x1      │ x2      │ x3      │
│     │ Float64 │ Float64 │ Float64 │
├─────┼─────────┼─────────┼─────────┤
│ 1   │ 1.0     │ 1.0     │ 1.0     │
│ 2   │ 1.0     │ 1.0     │ 1.0     │

julia> df[1, :] .= [1, 2, 3]
DataFrameRow
│ Row │ x1      │ x2      │ x3      │
│     │ Float64 │ Float64 │ Float64 │
├─────┼─────────┼─────────┼─────────┤
│ 1   │ 1.0     │ 2.0     │ 3.0     │
```
but we do not allow:
```
julia> df[1, :] .= [1 2 3]
ERROR: DimensionMismatch("cannot broadcast array to have fewer dimensions")
```
but we allow (as `1:1` does not drop a dimension):
```
julia> df[1:1, :] .= [1 2 3]
1×3 SubDataFrame
│ Row │ x1      │ x2      │ x3      │
│     │ Float64 │ Float64 │ Float64 │
├─────┼─────────┼─────────┼─────────┤
│ 1   │ 1.0     │ 2.0     │ 3.0     │
```

Currently we do not allow:
```
julia> df[1, :] .= df[1, :]
ERROR: ArgumentError: broadcasting over `DataFrameRow`s is reserved
```
(i.e. for `DataFrameRow` to be on RHS of broadcasted assignment, as we are waiting for Base to decide how to treat `NamedTuple` in the same place).

In summary: all the infrastructure of DataFrames.jl for `DataFrameRow` assumes it is 1-dimensional.

Reply:
OK, I guess that's fine then.